src/Snowflake/Command.php

```diff
@@ -149,18 +226,270 @@ public static function cloneDatabaseFromShared(Connection $connection, array $da
+            }
+        }
+        return $tmp;
+    }
+
+    private static function assignWorkspacesUsersAndRoles(Connection $connection, array $roles, array $users): array
+    {
+        foreach ($roles as $role) {
+            $grantsToRole = $connection->fetchAll(sprintf(
+                'SHOW GRANTS TO ROLE %s;',
+                $role['name']
+            ));
+
+            $ownershipToRole = array_filter($grantsToRole, fn($v) => $v['privilege'] === 'OWNERSHIP');
+
+            $rolesInRole = array_filter($ownershipToRole, fn($v) => $v['granted_on'] === 'ROLE');
+            $filteredRolesInRole = array_filter($rolesInRole, fn($v) => !in_array($v['name'], $roles));
+            $roles = array_merge($roles, array_combine(
+                array_map(fn($v) => $v['name'], $filteredRolesInRole),
+                $filteredRolesInRole
+            ));
+
+            $usersInRole = array_filter($ownershipToRole, fn($v) => $v['granted_on'] === 'USER');
+            $filteredUsersInRole = array_filter($usersInRole, fn($v) => !in_array($v['name'], $users));
+            $users = array_merge($users, array_map(fn($v) => $v['name'], $filteredUsersInRole));
+        }
+
+        return ['roles' => $roles, 'users' => $users];
+    }
+
+    public static function grantRoleToUser(Connection $connection, string $user, string $role): void
+    {
+        $connection->query(sprintf(
+            'GRANT ROLE %s TO USER %s',
+            QueryBuilder::quoteIdentifier($role),
+            QueryBuilder::quoteIdentifier($user)
+        ));
+    }
+
+    public static function assignGrantToRole(Connection $connection, array $grant): void
+    {
+        self::useRole($connection, $grant['granted_by']);
+
+        if ($grant['privilege'] === 'USAGE' && $grant['granted_on'] === 'ROLE') {
+            $sql = sprintf(
+                'GRANT %s %s TO %s %s %s',
+                $grant['granted_on'],
+                $grant['name'],
+                $grant['granted_to'],
+                $grant['grantee_name'],
+                $grant['grant_option'] === 'true' ? 'WITH GRANT OPTION' : '',
+            );
+        } else {
+            $sql = sprintf(
+                'GRANT %s ON %s %s TO %s %s %s',
+                $grant['privilege'],
+                $grant['granted_on'],
+                $grant['granted_on'] !== 'ACCOUNT' ? $grant['name'] : '',
+                $grant['granted_to'],
+                $grant['grantee_name'],
+                $grant['grant_option'] === 'true' ? 'WITH GRANT OPTION' : '',
+            );
+        }
+
+        $connection->query($sql);
+    }
+
+    public static function createMainRole(
+        Connection $connection,
+        string $warehouse,
+        array $users
+    ): void {
+        $mainRole = 'KEBOOLA_STORAGE';
+        $user = $mainRole;
+
+        self::createRole($connection, ['name' => $mainRole]);
+
+        $connection->query(sprintf(
+            'DROP WAREHOUSE IF EXISTS %s',
+            $warehouse
+        ));
```

Comment:
😱

Reply:
jo to jsem měl než jsem udělal cleanup metodu... dám pryč 👍

Reply:
https://github.com/keboola/project-migration-tool/pull/1/commits/5d3c81029d703c6e34917f1f99e24e8fb12b913e